cloud/scope/object_storage_bucket.go

```diff
@@ -90,6 +90,15 @@ func (s *ObjectStorageBucketScope) AddFinalizer(ctx context.Context) error {
 	return nil
 }
 
+// RemoveFinalizer removes a finalizer immediately patches the
+// object to avoid any race conditions.
+func (s *ObjectStorageBucketScope) RemoveFinalizer(ctx context.Context) error {
+	if controllerutil.RemoveFinalizer(s.Object, infrav1alpha1.GroupVersion.String()) {
+		return s.Close(ctx)
+	}
+	return nil
+}
+
```

Comment:
I see that the other controllers are calling `controllerutil.RemoveFinalizer` directly. IMO we should be somewhat consistent. I wouldn't mind having all scopes implement `RemoveFinalizer` since they all have `AddFinalizer`. We could also make a separate helper for it and add it as an inline struct, or maybe require that all scopes implement some "`FinalizerOps`" interface. What are your thoughts?